cmd/builder/internal/builder/config.go

```diff
@@ -34,8 +34,9 @@ var ErrInvalidGoMod = errors.New("invalid gomod specification for module")
 // Config holds the builder's configuration
 type Config struct {
 	Logger          *zap.Logger
-	SkipCompilation bool `mapstructure:"-"`
-	SkipGetModules  bool `mapstructure:"-"`
+	SkipCompilation bool   `mapstructure:"-"`
+	SkipGetModules  bool   `mapstructure:"-"`
+	LDFlags         string `mapstructure:"-"`
```

Comment:
How do you expect this to be configured?

Reply:
As a command line argument. Ex:
```
builder --ldflags='-X "<pkg>.gitVersion=d49d45c039a2a89a26526afa3658de6429099bef" -X "<pkg>.gitTag=local-testing" -X "<pkg>.whoami=kristina.pathak"'
```

Reply:
where is the "--ldflags" flag defined?

Reply:
In `cmd/builder/internal/command.go`:
https://github.com/open-telemetry/opentelemetry-collector/pull/6946/files#diff-6019321caab5c53bd0ae7d59974bf883a6101939b98e3ca4d585c7018fa34207R90